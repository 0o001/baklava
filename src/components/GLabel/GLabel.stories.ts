import { storiesOf } from '@storybook/vue';
import { withKnobs, text } from '@storybook/addon-knobs';
import GLabel from './GLabel.vue';

const stories = storiesOf('GLabel', module);
stories.addDecorator(withKnobs);

stories
  .addParameters({ component: GLabel })
  .add('base', () => ({
    components: { GLabel },
    props: {
      iconName: {
        default: text('Icon Name', 'alert-circle'),
      },
    },
    template: `<GLabel :iconName="iconName" variant="base">Base Label</GLabel>`,
  }))
  .add('success', () => ({
    components: { GLabel },
    props: {
      iconName: {
        default: text('Icon Name', 'alert-circle'),
      },
    },
    template: `<GLabel :iconName="iconName" variant="success">Success Label</GLabel>`,
  }))
  .add('warning', () => ({
    components: { GLabel },
    props: {
      iconName: {
        default: text('Icon Name', 'alert-circle'),
      },
    },
    template: `<GLabel :iconName="iconName" variant="warning">Warning Label</GLabel>`,
  }))
  .add('error', () => ({
    components: { GLabel },
    props: {
      iconName: {
        default: text('Icon Name', 'alert-circle'),
      },
    },
    template: `<GLabel :iconName="iconName" variant="error">Error Label</GLabel>`,
  }))
