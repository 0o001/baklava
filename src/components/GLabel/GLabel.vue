<template>
  <span class="g-label">
    <GIcon
      class="g-mr-10"
      size="20px"
      :name="iconName"
      :color="textColor"
    />
    <GText
      variant="subtitle-03"
      :color="textColor"
    >
      <slot />
    </GText>
  </span>
</template>

<script lang="ts">

import GIcon from '../GIcon/GIcon.vue';
import GText from '../GText/GText.vue';

export default {
  name: 'GLabel',
  components: {
    GIcon,
    GText,
  },
  props: {
    iconName: {
      type: String,
      default: 'alert-circle',
    },
    variant: {
      type: String,
      default: 'base',
    },
  },
  computed: {
    textColor () {
      return {
        base: 'dark-grey-500',
        success: 'green-500',
        warning: 'orange-500',
        error: 'red-500',
      }[this.variant];
    },
  },
};
</script>

<style lang="scss" scoped>
.g-label {
  display: flex;
  align-items: center;
  font-family: var(--font-family-base);
  line-height: normal;
}
</style>
