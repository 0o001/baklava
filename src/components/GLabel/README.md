# GLabel

## Slots

| Name      | Description |
| --------- | ----------- |
| `default` | &nbsp;      |

## Props

| Name        | Type     | Description | Default          |
| ----------- | -------- | ----------- | ---------------- |
| `icon-name` | `String` |             | `"alert-circle"` |
| `variant`   | `String` |             | `"base"`         |

## Computed Properties

| Name        | Type     | Description                 |
| ----------- | -------- | --------------------------- |
| `textColor` | `object` | **Dependencies:** `variant` |

