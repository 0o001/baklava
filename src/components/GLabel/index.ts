import GLabel from './GLabel.vue';

export default GLabel;
