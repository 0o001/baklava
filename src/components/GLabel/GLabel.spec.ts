import { shallowMount } from '@vue/test-utils';
import GLabel from './';

describe('GLabel', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(GLabel);
  });

  it('should match snapshot when show is prop is given as false ', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render base variant as default when variant prop does not set', () => {
    expect(wrapper.vm.textColor).toEqual('dark-grey-500');
  });

  it('should render given variant when variant prop set', async () => {
    // given
    wrapper.setProps({
      variant: 'success',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.textColor).toEqual('green-500');
  });
});
