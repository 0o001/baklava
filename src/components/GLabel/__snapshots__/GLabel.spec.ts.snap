// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GLabel should match snapshot when show is prop is given as false  1`] = `
<span
  class="g-label"
>
  <gicon-stub
    class="g-mr-10"
    color="dark-grey-500"
    fill="none"
    name="alert-circle"
    size="20px"
    stroke="currentColor"
    strokelinecap="round"
    strokelinejoin="round"
    strokewidth="2"
    tag="i"
  />
   
  <gtext-stub
    color="dark-grey-500"
    variant="subtitle-03"
  />
</span>
`;
